tests/integration_tests/style/test_repo.py

```diff
@@ -34,3 +35,25 @@ def test_repo_validate_yaml():
         if str(path).startswith("../build/"):
             continue
         yaml.safe_load(path.open(encoding="utf-8"))
+
+
+def test_repo_validate_changelog():
+    """Make sure the CHANGELOG.md file follows the Keep a Changelog format"""
+
+    changelog_path = Path("../CHANGELOG.md")
+    changelog = changelog_path.read_text().splitlines()
+    errors = []
+    for lineno, line in enumerate(changelog, start=1):
+        if line.startswith("## "):
+            if not re.match(r"^## \[.+\]$", line):
+                msg = "Level 2 headings (versions) should be wrapped in []"
+                errors.append((lineno, msg, line))
+        if line.startswith("### "):
+            if not re.match(r"^### (Added|Changed|Deprecated|Removed|Fixed)$", line):
+                msg = "Unknown Level 3 heading"
+                errors.append((lineno, msg, line))
+
+    for lineno, msg, line in errors:
+        print(msg)
+        print(f"\t{lineno}:{line}")
+    assert len(errors) == 0
```

Comment:
I agree if it was a single error. With several errors, the output just didn't look as nice :)

Reply:
I remember I was [breeding monsters](https://github.com/firecracker-microvm/firecracker/blob/main/tests/integration_tests/functional/test_cpu_features.py#L452) for that, so it can be preformatted, but I agree it may not be worth it. Just only if the print is surpassed due to a change in pytest settings somehow.

Reply:
Yeah I think that pytest assert does a great job of diffing objects when it's a list/set/dict, but a list of strings may not be much benefit.. I'm not married to it though, if this ends up being more confusing than helping we can revert it.